packages/app/components/Settings/CourseRoster.tsx

```diff
@@ -0,0 +1,122 @@
+import { SearchOutlined } from "@ant-design/icons";
+import { API } from "@koh/api-client";
+import { Role } from "@koh/common";
+import { Divider, Input, List, Pagination, Spin } from "antd";
+import Avatar from "antd/lib/avatar/avatar";
+import { useState } from "react";
+import { ReactElement } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+type CourseRosterProps = { courseId: number };
+
+type RenderTableProps = {
+  courseId: number;
+  role: Role;
+  listTitle: string;
+  displaySearchBar: boolean;
+  searchPlaceholder: string;
+};
+
+const CourseRosterComponent = styled.div`
+  margin-left: auto;
+  margin-right: auto;
+`;
+
+export default function CourseRoster({
+  courseId,
+}: CourseRosterProps): ReactElement {
+  return (
+    <CourseRosterComponent>
+      <h1>Course Roster</h1>
+      <RenderTable
+        courseId={courseId}
+        role={Role.PROFESSOR}
+        listTitle={"Professors"}
+        displaySearchBar={false}
+        searchPlaceholder=""
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.TA}
+        listTitle={"Teaching Assistants"}
+        displaySearchBar={true}
+        searchPlaceholder="Search TAs"
+      />
+      <br />
+      <RenderTable
+        courseId={courseId}
+        role={Role.STUDENT}
+        listTitle={"Students"}
+        displaySearchBar={true}
+        searchPlaceholder="Search students"
+      />
+      <br />
+      <Divider />
+    </CourseRosterComponent>
+  );
+}
+
+function RenderTable({
+  courseId,
+  role,
+  listTitle,
+  displaySearchBar,
+  searchPlaceholder,
+}: RenderTableProps): ReactElement {
+  const [page, setPage] = useState(1);
+  const [search, setSearch] = useState("");
+  const handleSearch = (event) => {
+    event.preventDefault();
+    setSearch(event.target.value);
+    setPage(1);
+  };
+  const { data } = useSWR(
+    `${role}/${page}/${search}`,
+    async () => await API.course.getUserInfo(courseId, page, role, search)
+  );
+  if (!data) {
+    return <Spin tip="Loading..." size="large" />;
+  } else {
+    return (
+      <div style={{ backgroundColor: "white" }}>
+        <div style={{ backgroundColor: "#f0f0f0", height: "56px" }}>
+          <h3 style={{ position: "relative", left: "10px", top: "14px" }}>
```

Comment:
might be nice to put some of these as styled components :>

Reply:
Got 1 styled component in, the second div doesn't keep its color when I try to change it to a styled component so I think I'll keep that as is